dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -42,10 +42,45 @@ def login
     redirect_to auth_redirect_url.to_s
   end
 
+  def authenticate
+    id_token = params[:id_token]
+    return unauthorized_status unless id_token
+    begin
+      decoded_jwt_no_auth = JSON::JWT.decode(id_token, :skip_verification)
+    rescue
+      return unauthorized_status
+    end
+    # client_id is the aud[ience] in the JWT
+    extracted_client_id = decoded_jwt_no_auth[:aud]
+    extracted_issuer_id = decoded_jwt_no_auth[:iss]
+
+    integration = LtiIntegration.find_by({client_id: extracted_client_id, issuer: extracted_issuer_id})
+    return unauthorized_status unless integration
+
+    # verify the jwt via the integration's public keyset
+    decoded_jwt = get_decoded_jwt(integration, id_token)
+
+    jwt_verifier = JwtVerifier.new(decoded_jwt, integration)
+
+    if jwt_verifier.verify_jwt
+      target_link_uri = decoded_jwt_no_auth[:'https://purl.imsglobal.org/spec/lti/claim/target_link_uri']
+      redirect_to target_link_uri
+    else
+      return unauthorized_status(*jwt_verifier.errors)
```

Comment:
Thinking through security implications of returning the specifics of the error here, vs. a standard 'unauthorized'. We don't want to give the caller too much information about why their call failed. I think we should log these errors, to help us debug problems internally, but not return them in the response. This [jira ticket](https://codedotorg.atlassian.net/browse/P20-115) is for tracking follow up logging/metrics for this new controller.

Reply:
Keeping them was helpful for debugging the tests and trying to hit all of the corner cases, but I can change it up to only give the unauthorized status.